easybuild/easyblocks/p/pytorch.py

```diff
@@ -155,29 +155,35 @@ def configure_step(self):
-
-        nccl_root = get_software_root('NCCL')
-        if nccl_root:
-            options.append('USE_SYSTEM_NCCL=1')
-            options.append('NCCL_INCLUDE_DIR=' + os.path.join(nccl_root, 'include'))
-
-        # list of CUDA compute capabilities to use can be specifed in two ways (where (2) overrules (1)):
-        # (1) in the easyconfig file, via the custom cuda_compute_capabilities;
-        # (2) in the EasyBuild configuration, via --cuda-compute-capabilities configuration option;
-        cuda_cc = build_option('cuda_compute_capabilities') or self.cfg['cuda_compute_capabilities']
-        if not cuda_cc:
-            raise EasyBuildError("List of CUDA compute capabilities must be specified, either via " +
-                                 "cuda_compute_capabilities easyconfig parameter or via --cuda-compute-capabilities")
-        self.log.info("Compiling with specified list of CUDA compute capabilities: %s", ', '.join(cuda_cc))
-        options.append('TORCH_CUDA_ARCH_LIST="%s"' % ';'.join(cuda_cc))
+        if get_software_root('CUDA'):
+            options.append('USE_CUDA=1')
+            cudnn_root = get_software_root('cuDNN')
+            if cudnn_root:
+                options.append('CUDNN_LIB_DIR=' + os.path.join(cudnn_root, 'lib64'))
+                options.append('CUDNN_INCLUDE_DIR=' + os.path.join(cudnn_root, 'include'))
+
+            nccl_root = get_software_root('NCCL')
+            if nccl_root:
+                options.append('USE_SYSTEM_NCCL=1')
+                options.append('NCCL_INCLUDE_DIR=' + os.path.join(nccl_root, 'include'))
+
+            # list of CUDA compute capabilities to use can be specifed in two ways (where (2) overrules (1)):
+            # (1) in the easyconfig file, via the custom cuda_compute_capabilities;
+            # (2) in the EasyBuild configuration, via --cuda-compute-capabilities configuration option;
+            cuda_cc = build_option('cuda_compute_capabilities') or self.cfg['cuda_compute_capabilities']
+            if not cuda_cc:
+                raise EasyBuildError('List of CUDA compute capabilities must be specified, either via '
+                                     'cuda_compute_capabilities easyconfig parameter or via '
+                                     '--cuda-compute-capabilities')
+            self.log.info('Compiling with specified list of CUDA compute capabilities: %s', ', '.join(cuda_cc))
+            options.append('TORCH_CUDA_ARCH_LIST="%s"' % ';'.join(cuda_cc))
+        else:
+            # Disable CUDA
+            options.append('USE_CUDA=0')
 
         if get_cpu_architecture() == POWER:
             # *NNPACK is not supported on Power, disable to avoid warnings
-            nnpacks = ('NNPACK', 'QNNPACK', 'USE_PYTORCH_QNNPACK', 'USE_XNNPACK')
+            nnpacks = ('NNPACK', 'QNNPACK', 'PYTORCH_QNNPACK', 'XNNPACK')
             options.extend('USE_%s=0' % nnpack for nnpack in nnpacks)
```

Comment:
@Flamefire As you suggested, it's probably better to not use the prefix approach, but make it something more straightforward that is easy to `grep`, like:

```python
options.extend(['USE_NNPACK=0', 'USE_QNNPACK=0', 'USE_PYTORCH_QNNPACK=0', 'USE_XNNPACK=0'])

Reply:
Done https://github.com/easybuilders/easybuild-easyblocks/pull/2164/commits/ae7f815252333b68c5955434a23bb09975f6078e